crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -620,12 +623,30 @@ impl SpecializedMeshPipeline for MeshPipeline {
         let vertex_buffer_layout = layout.get_layout(&vertex_attributes)?;
 
         let (label, blend, depth_write_enabled);
-        if key.contains(MeshPipelineKey::TRANSPARENT_MAIN_PASS) {
-            label = "transparent_mesh_pipeline".into();
-            blend = Some(BlendState::ALPHA_BLENDING);
+        let pass = key.intersection(MeshPipelineKey::BLEND_RESERVED_BITS);
+        if pass == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA {
+            label = "premultiplied_alpha_mesh_pipeline".into();
+            blend = Some(BlendState::PREMULTIPLIED_ALPHA_BLENDING);
+            shader_defs.push("PREMULTIPLY_ALPHA".to_string());
+            shader_defs.push("BLEND_PREMULTIPLIED_ALPHA".to_string());
             // For the transparent pass, fragments that are closer will be alpha blended
             // but their depth is not written to the depth buffer
             depth_write_enabled = false;
+        } else if pass == MeshPipelineKey::BLEND_MULTIPLY {
+            label = "multiply_mesh_pipeline".into();
+            blend = Some(BlendState {
+                color: BlendComponent {
+                    src_factor: BlendFactor::Dst,
+                    dst_factor: BlendFactor::OneMinusSrcAlpha,
```

Comment:
Can we expose the blend component to shader somehow in a custom blend mode?

Reply:
While exposing `BlendComponent`, `BlendFactor` and `BlendOperation` (like Unity seems to do) directly via the `StandardMaterial`  _could_ be done:

- To support the full range of combinations, it would require using at least 11 additional bits of the `MeshPipelineKey`; (Instead of just two bits for the current approach) — and up to 22 bits if you wanted to also support the alpha options.
- Most of the possible combinations are either meaningless or not very useful when combined with the PBR `StandardMaterial`.

So I'm not sure if it makes sense, but would like others to chime in.

One thing we _could do_ (but ideally as a separate PR) is to introduce a `fn blend_state() -> Option<BlendState>` method to the `Material` trait, so that we don't have to use the [somewhat large specialization code snippet](https://github.com/bevyengine/bevy/pull/6644#issuecomment-1321413203) I left on my other comment.

Reply:
It would be nice to have an option to use it, rather than gate it arbitrarily, in my case I needed it.

I would much rather have the user pick what options they need than have a few preselected ones

Reply:
I think a custom material is simple enough that I would suggest users do that if they need more unusual behavior.